Update glfw submodule to latest

SaschaWillems opened the pull request:
## Description

This PR updates the glfw submodule to the latest release. We haven't updated this in ages, which seems to cause issues for some newer linux distros.

Did two full batch runs (debug and release build) in Windows to make sure changes don't cause any issues. 

Fixes #1192

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)

 If this PR contains framework changes:
 - [x] I did a full batch run using the `batch` command line argument to make sure all samples still work properly

SaschaWillems committed:
Update glfw submodule to latest

Old version was 3.3, released over two years ago
Fixes #1192

SaschaWillems committed:
Install libraries required by new GLFW version

SaschaWillems committed:
Fixing library names

SaschaWillems committed:
Fixing type

SaschaWillems committed:
Also add new GLFW required libs to v2 yml

SaschaWillems committed:
Also add new libraries to documentation ci step
Weird, but somehow required

SaschaWillems committed:
Update linux build docs with new libs required by updated GLFW

SaschaWillems commented:
@tomadamatkinson : Clang Tidy Check fails because glfw needs new libraries. Any idea why that needs libraries at all? Doesn't it just check code files? I added new required libs to all steps that require them, but I can't see where those need to be set up for Clang Tidy.

P.S. : I think we need to clean up / streamline CI. So many places that you have to care for when new requirements come in :/

tomadamatkinson commented:
There is a PR which merges v2 with the existing pipeline and removes the Debug builds. That should help.

In terms of these dependencies I think they are now default required by GLFW. You can turn off Wayland builds with a CMake flag. Not sure why Antora needs it... is that not a documentation pass

SaschaWillems commented:
Well, the problem is that EVERY CI step that does build samples needs the new libraries. I already added them to all CI steps incl. Antora, but I have no clue how to add them to the Clang Tidy step. It's the last one failing, so I need some help here ;)

SaschaWillems commented:
@tomadamatkinson : I could use a hand with this ;) I don't know what I can do to fix the failing Clang Tidy Check.

gpx1000 commented:
I'll try to take a look.

On Wed, Dec 11, 2024, 8:34 AM Sascha Willems ***@***.***>
wrote:

> @tomadamatkinson <https://github.com/tomadamatkinson> : I could use a
> hand with this ;) I don't know what I can do to fix the failing Clang Tidy
> Check.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/KhronosGroup/Vulkan-Samples/pull/1237#issuecomment-2536496236>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5IAYYKNQ2UWVU3A5RTG3T2FBSQDAVCNFSM6AAAAABSYPQMKOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMZWGQ4TMMRTGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

gpx1000 committed:
Remove Wayland dependencies and conditionally build Antora

Wayland dependencies were removed from the workflows and build configurations to streamline the installation process. The Antora documentation build is now conditional, preventing unnecessary inclusion when VKB_GENERATE_ANTORA_SITE is not set. This improves build flexibility and reduces extraneous steps.

gpx1000 commented:
https://github.com/SaschaWillems/Vulkan-Samples/pull/5

I removed the wayland need for now in the CI by passing it -DGLFW_BUILD_WAYLAND=OFF from the command line in the CI.  Wayland in glfw is still a little half baked and not required to build / use GLFW.  All versions of GLFW prior to this one had Wayland building off by default.  We'll revisit this need as the dependencies like clang-tidy gain support in the future.  NB: this doesn't stop anyone from trying to build with Wayland if they have it.  Just disables it from our CI.  It *might* make sense to do some documentation so end users know they now need to have wayland-dev installed, but there's plenty of messaging about that from GLFW automatically as is; so probably not needed.

Oh, because it has been a request, I disabled the doc build from doing anything other than generate docs.  You no longer even need to download the submodules (I don't think).  However, you would no longer be able to build both docs and samples binaries in one go; don't think that use case is ever needed so no real loss there.

gpx1000 commented:
I should say, another way to "fix" this if you prefer is to simply fix the clang-tidy build. check.yml at line 101 you can add -DGLFW_BUILD_WAYLAND=OFF or, you could solve it the same way you did in the other build types.  just add a run line that adds wayland-dev etc apt repositories to the linux builder.
Up to you how you want to fix it ;-)  my PR passed though so should be good from there.

SaschaWillems committed:
Merge pull request #5 from gpx1000/for_sascha_update_glfw

Remove Wayland dependencies and conditionally build Antora

SaschaWillems committed:
Merge remote-tracking branch 'origin/main' into update_glfw

# Conflicts:
#	.github/workflows/v2.yml

SaschaWillems committed:
Apply required changes to merged CI changes

SaschaWillems committed:
Trying to fix CI after merge (again)

SaschaWillems committed:
Trying to fix CI after merge (again and again)
